cms/djangoapps/modulestore_migrator/tasks.py

```diff
@@ -127,56 +114,22 @@ class _MigrationContext:
     """
     Context for the migration process.
     """
-    existing_source_to_target_keys: dict[  # Note: It's intended to be mutable to reflect changes during migration.
-        UsageKey, list[PublishableEntity | None]
-    ]
+    # Fields that get mutated as we migrate blocks
+    used_component_keys: set[LibraryUsageLocatorV2]
+    used_container_slugs: set[str]
+
+    # Fields that remain constant
+    previous_block_migrations: dict[UsageKey, data.ModulestoreBlockMigrationResult]
```

Comment:
The simplest way to remain consistent around which set of previous migrations we're respecting (for the purpose of RepeatHandlingStrategy.Update and RepeatHandlingStrategy.Skip) is to load them up ahead of time.

(A potential future optimization, if it's necessary, is to not load up the dict at all if we're using `RepeatHandlingStrategy.Fork`)